src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -1714,8 +1714,15 @@ public ITextComponent getDisplayName(@NotNull final IToken token)
             return Optional.empty();
         }
 
-        Integer citizenData = requestsByCitizen.get(token);
-        return Optional.of(getColony().getCitizen(citizenData));
+        int citizenID = requestsByCitizen.get(token);
+
+        if(getColony().getCitizen(citizenID) == null)
```

Comment:
This is not needed. If you pass null to a Optional, it will create an Empty Optional

Reply:
how does it crash in the following line then with an NPE?

Reply:
https://github.com/Minecolonies/minecolonies/issues/1907

Reply:
getColony() is null